pkg/tfbridge/x/token_test.go

```diff
@@ -472,46 +473,333 @@ func TestAliasing(t *testing.T) {
+	err = AutoAliasing(info, metadata)
+	require.NoError(t, err)
+
+	assert.True(t, *info.Resources["pkg_r1"].Fields["f1"].MaxItemsOne)
+	assert.False(t, *info.Resources["pkg_r1"].Fields["f2"].MaxItemsOne)
+	assert.Equal(t, expected, string(metadata.Marshal()))
+
+	// Validate that overrides work
+
+	info = provider(true, false)
+	info.Resources["pkg_r1"].Fields = map[string]*tfbridge.SchemaInfo{
+		"f1": {MaxItemsOne: tfbridge.False()},
+	}
+
+	err = AutoAliasing(info, metadata)
+	require.NoError(t, err)
+	assert.False(t, *info.Resources["pkg_r1"].Fields["f1"].MaxItemsOne)
+	assert.False(t, *info.Resources["pkg_r1"].Fields["f2"].MaxItemsOne)
+	assert.Equal(t, `{
+    "auto-aliasing": {
+        "resources": {
+            "pkg_r1": {
+                "current": "pkg:index/r1:R1",
+                "fields": {
+                    "f1": {
+                        "maxItemOne": false
+                    },
+                    "f2": {
+                        "maxItemOne": false
+                    }
+                }
+            }
+        },
+        "datasources": {}
+    }
+}`, string(metadata.Marshal()))
+}
+
+func TestMaxItemsOneAliasingExpiring(t *testing.T) {
+	provider := func(f1, f2 bool) *tfbridge.ProviderInfo {
+		prov := &tfbridge.ProviderInfo{
+			P: Provider{
+				resources: map[string]shim.SchemaMap{
+					"pkg_r1": newSchemaMap(map[string]shim.Schema{
+						"f1": Schema{MaxItemsOne: f1},
+						"f2": Schema{MaxItemsOne: f2},
+					}),
+				},
+			},
+		}
+		err := ComputeDefaults(prov, TokensSingleModule("pkg_", "index", MakeStandardToken("pkg")))
+		require.NoError(t, err)
+		return prov
+	}
+	info := provider(true, false)
+	metadata, err := metadata.New(nil)
+	require.NoError(t, err)
+
+	// Save current state into metadata
+	err = AutoAliasing(info, metadata)
+	require.NoError(t, err)
+
+	v := string(metadata.Marshal())
+	expected := `{
+    "auto-aliasing": {
+        "resources": {
+            "pkg_r1": {
+                "current": "pkg:index/r1:R1",
+                "fields": {
+                    "f1": {
+                        "maxItemOne": true
+                    },
+                    "f2": {
+                        "maxItemOne": false
+                    }
+                }
+            }
+        },
+        "datasources": {}
+    }
+}`
+	assert.Equal(t, expected, v)
+
+	info = provider(false, true)
+
+	// Apply metadata back into the provider
+	info.Version = "1.0.0" // New major version
+	err = AutoAliasing(info, metadata)
+	require.NoError(t, err)
+
+	assert.Nil(t, info.Resources["pkg_r1"].Fields["f1"].MaxItemsOne)
+	assert.Nil(t, info.Resources["pkg_r1"].Fields["f2"].MaxItemsOne)
+	assert.Equal(t, `{
+    "auto-aliasing": {
+        "resources": {
+            "pkg_r1": {
+                "current": "pkg:index/r1:R1",
+                "majorVersion": 1,
+                "fields": {
+                    "f1": {
+                        "maxItemOne": false
```

Comment:
Perhaps `omitempty` to skip false ones in the serial form.

Reply:
The underlying type is a `*bool`. We actually have `omitempty` applied, but we need to keep track of previous values, both true and false.